middleware/lib/utils/s3.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const S3 = require('aws-sdk/clients/s3');
+
+const s3 = new S3();
+
+/**
+ * Given a VIN return the bucket name
+ *
+ * @param {string} vin - The VIN
+ */
+const getBucketName = vin => `ford-capstone-wayne-state/${vin}`;
+
+/**
+ * @typedef {import('aws-sdk').AWSError} AWSError
+ */
+
+/**
+ * Delete a single file from s3.
+ *
+ * @param {string} vin  - The VIN associated with the file.
+ * @param {string} name - The name of the file to delete.
+ */
+exports.deleteFile = (vin, name) => {
+  return s3
+    .deleteObject({
+      Bucket: getBucketName(vin),
+      Key: name,
+    })
+    .promise();
+};
+
+/**
+ * Get a single file from S3 or throw a 412 "Precondition Failed" if hashes don't match.
+ *
+ * @param {string} vin  - The VIN associated with the file.
+ * @param {string} name - The filename.
+ * @param {string} hash - The file's hash from the blockchain.
+ * @throws {AWSError} Error with `statusCode` 412 if there is a mismatch in hashes.
```

Comment:
## THIS IS THE MOST IMPORTANT THING TO NOTE

This function will throw an error with a `412` statusCode if the file's hash in s3 does not match the hash that is saved on the blockchain.